practice-two/src/App.jsx

```diff
@@ -104,14 +104,21 @@ const App = () => {
    * @returns {Array} - Updated list of products cart IDs
    */
   const addToCart = (productID) => {
-    setCartProductIDs((prevCartIDs) => {
-      // If the product is already in the cart, don't add it again
-      if (prevCartIDs.includes(productID)) {
-        return prevCartIDs;
+    setCartProductIDs((prevCartProducts) => {
+      const existingProduct = prevCartProducts.find((item) => item.id === productID);
+
+      if (existingProduct) {
+        setShowAddToCartModal(true);
+        // If the product already exists in the cart, update its quantity
+        return prevCartProducts.map((item) =>
+          item.id === productID ? { ...item, quantity: item.quantity + 1 } : item,
+        );
       }
+
```

Comment:
Why `setShowAddToCartModal(true);`  twice ?

![image](https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/27277174/07272a87-b96d-4c6a-896a-5bb4d0f6ae66)

Reply:
setShowAddToCartModal displays a message that the user has added successfully, and in the add function there are 2 states that the product is already in the list and the product is not in the list. When successfully added, call setShowAddToCartModal to display the Notification window